pkg/releasepayload/jobrunresult/helpers.go

```diff
@@ -16,3 +19,129 @@ func (in ByCoordinatesName) Len() int {
 func (in ByCoordinatesName) Swap(i, j int) {
 	in[i], in[j] = in[j], in[i]
 }
+
+func SetJobRunResult(results *[]v1alpha1.JobRunResult, newResult v1alpha1.JobRunResult) {
+	if results == nil {
+		results = &[]v1alpha1.JobRunResult{}
+	}
+	existingResult := FindJobRunResult(*results, newResult.Coordinates)
+	if existingResult == nil {
+		*results = append(*results, newResult)
+		return
+	}
+
+	existingResult.CompletionTime = newResult.CompletionTime
+	existingResult.HumanProwResultsURL = newResult.HumanProwResultsURL
+	existingResult.StartTime = newResult.StartTime
+	existingResult.State = newResult.State
+}
+
+func RemoveJobRunResult(results *[]v1alpha1.JobRunResult, coordinates v1alpha1.JobRunCoordinates) {
+	if results == nil {
+		results = &[]v1alpha1.JobRunResult{}
+	}
+	newResults := []v1alpha1.JobRunResult{}
+	for _, result := range *results {
+		if !cmp.Equal(result.Coordinates, coordinates) {
+			newResults = append(newResults, result)
+		}
+	}
+
+	*results = newResults
+}
+
+func FindJobRunResult(results []v1alpha1.JobRunResult, coordinates v1alpha1.JobRunCoordinates) *v1alpha1.JobRunResult {
+	for i := range results {
+		if cmp.Equal(results[i].Coordinates, coordinates) {
+			return &results[i]
+		}
+	}
+
+	return nil
+}
+
+func ComputeAggregateState(job v1alpha1.JobStatus) v1alpha1.JobState {
```

Comment:
single call site?  If so, put it in that controller file.

Reply:
Done